feat: AKS add support for spot node pool upgrades

weisdd opened the pull request:
Starting from June, AKS supports spot node pool upgrades:
- https://github.com/Azure/AKS/blob/master/CHANGELOG.md#release-2022-06-12
- https://docs.microsoft.com/en-gb/azure/aks/spot-node-pool#upgrade-a-spot-node-pool

This PR removes the respective restriction from the provider code. No library updates are needed. 

Fixes: #14786

stephybun reviewed (changes requested):
Hey @weisdd,

thanks for updating the resource to support this feature, could we add a test for this as well to ensure it works?

weisdd commented:
Hey @stephybun,

Sure. I've just rebased the branch to incorporate the latest provider code and added the upgrade test.

Implementation details:
- The test is similar to `TestAccKubernetesCluster_upgradeControlPlaneAndAllPoolsTogetherVersionAlias`.
- terraform-code is generated through `upgradeVersionsConfigSpot` to avoid tests relying on `upgradeVersionsConfig` from passing too many parameters to it. Please, let me know if you prefer a different approach.

```shell
$ make acctests SERVICE='containers' TESTARGS='-run=TestAccKubernetesCluster_upgradeControlPlaneAndAllPoolsTogetherSpot' TESTTIMEOUT='60m'

==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/containers -run=TestAccKubernetesCluster_upgradeControlPlaneAndAllPoolsTogetherSpot -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccKubernetesCluster_upgradeControlPlaneAndAllPoolsTogetherSpot
=== PAUSE TestAccKubernetesCluster_upgradeControlPlaneAndAllPoolsTogetherSpot
=== CONT  TestAccKubernetesCluster_upgradeControlPlaneAndAllPoolsTogetherSpot
--- PASS: TestAccKubernetesCluster_upgradeControlPlaneAndAllPoolsTogetherSpot (1756.83s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/containers    1757.975s
```

stephybun reviewed (approved):
Thanks @weisdd, LGTM 🍰
